Remove LazyProvide from credential provider

tiffanyfay opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/release.md#issue-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR strips out Lazy Provide from credential provider and kubelet. It was previously added in credential provider for AWS and now, none of the providers use Lazy Provide. 

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes https://github.com/kubernetes/kubernetes/issues/76465

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```
/sig aws
/sig cloud-provider
/sig node
/area cloudprovider
/area kubelet


@mcrute  @andrewsykim

Pingan2017 committed:
clean up redundant conditiontype OutOfDisk

Commit:
fix cp time stamp is in the future error

sjenning committed:
kubelet: add allowed sysctl to KubeletConfiguration

smarterclayton committed:
Add a metadata client to client-go that can read PartialObjectMetadata

This client exposes operations on generic metadata (get, list, watch, delete)
and allows patch operations. The client always uses protobuf and requests
the server transform the response into the appropriate object. Using this
client simplifies the work of generic controllers by allowing them to treat
all objects the same, and also improves performance both in the amount of
data sent as well as allowing protobuf on CRD resources.

andrewsykim commented on pkg/credentialprovider/provider.go:
```diff
@@ -37,22 +37,6 @@ type DockerConfigProvider interface {
-
-//LazyProvide returns an Lazy AuthConfig
-func LazyProvide(creds LazyAuthConfiguration, image string) AuthConfig {
-	if creds.Provider != nil {
```
So is it safe to assume Provider is not nil here anymore?

mcrute replied:
`Provider` should no longer exist in `creds` because that was the mechanism by which the lazy provider operated. With the previous patch `creds` is a real credential.

MaysaMacedo committed:
Increase Network Policy tests coverage

khenidak committed:
cri types changes

johscheuer committed:
Remove experimental note for ipvs in kube-proxy arguments

khenidak committed:
node ipam controller for ipv6 dualstack

khenidak committed:
route controller + azure v6 routes

odinuge committed:
Fix file permissions for non executable files

Normal files should have permissions 644 by default,
and does not require the last bit to be
executable

Signed-off-by: Odin Ugedal <odin@ugedal.com>

khenidak committed:
updated publishing rules

johnSchnake committed:
Adds an optional golang runner to the conformance test image

Adds a go app which runs the e2e tests with ginkgo.

 - Supports all the existing env vars of the bash script
 - Improved flow control to avoid and better report issues
regarding the process PID
 - Adds flags for modifying where to find the test binary and
ginkgo binary so that you can run it locally
 - Adds 3 flags for specifying extra args before the double-dash,
extra args after the double-dash, and the seperator to use between
values in those env vars. This allows setting arbitrary, complex
values for use on the command such as flags which include spaces
or other characters.

khenidak committed:
api: dropDisabledFields

khenidak committed:
IPv6DualStack ALPHA feature

khenidak committed:
types modifications + conversion + conversion testing

SataQiu committed:
kubeadm: add --control-plane-endpoint flag

wojtek-t committed:
Fix client-go request benchmark

jb-perez committed:
release lib: save release docker tag in tarfile

andyzhangx committed:
fix: change timeout value in csi plugin

khenidak committed:
vendor updates

danwinship committed:
Hack PatchNodeStatus() to override the patch type on Status.Addresses

khenidak committed:
generated items

tedyu committed:
Utilize RWMutex for efficient backoff operations

tedyu committed:
Omit obtaining the lock when adding Nodes in newNodeTree

khenidak committed:
kubenet for ipv6 dualstack

Commit:
correct the CHANGELOG-1.15.md

k8s-ci-robot committed:
Merge pull request #79642 from tedyu/new-node-tree

Omit obtaining the lock when adding Nodes in newNodeTree

k8s-ci-robot committed:
Merge pull request #79650 from qingsenLi/k8s-190702-fix-1.15

correct the CHANGELOG-1.15.md

k8s-ci-robot committed:
Merge pull request #78177 from MaysaMacedo/add-more-e2e-np-tests

Increase Network Policy tests coverage

k8s-ci-robot committed:
Merge pull request #79497 from javier-b-perez/release-lib

Save docker tag for releases in tarfile archive

k8s-ci-robot committed:
Merge pull request #79646 from tedyu/backoff-rw

Utilize RWMutex for efficient backoff operations

danwinship committed:
Document the problem with Node.Status.Addresses and strategic merge patch

k8s-ci-robot committed:
Merge pull request #79284 from johnSchnake/conformanceGoRunner

Adds an optional golang runner to the conformance test image

mcrute commented on pkg/kubelet/kuberuntime/kuberuntime_image.go:
```diff
@@ -17,11 +17,10 @@ limitations under the License.
 
 import (
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
:smirk:

tiffanyfay replied:
>.<

mcrute commented on pkg/credentialprovider/keyring.go:
```diff
@@ -36,18 +36,19 @@ import (
-// lazyDockerKeyring is an implementation of DockerKeyring that lazily
+// TODO  edit
+// MultiProviderDockerKeyring is an implementation of DockerKeyring that lazily
 // materializes its dockercfg based on a set of dockerConfigProviders.
```
I think this comment is fine I'd just remove the "lazily" bit from it

mcrute reviewed (commented):
I think this looks good and cleans up the old lazy provider stuff. I saw this work in a test cluster with ECR so I'm satisfied that this is safe to ship upstream. WDYT @andrewsykim?

andrewsykim commented on pkg/credentialprovider/keyring.go:
```diff
@@ -36,18 +36,19 @@ import (
+// MultiProviderDockerKeyring is an implementation of DockerKeyring that lazily
 // materializes its dockercfg based on a set of dockerConfigProviders.
-type lazyDockerKeyring struct {
+type MultiProviderDockerKeyring struct {
```
I think `ProvidersDockerKeyring` is okay too

tiffanyfay committed:
Remove LazyProvide from kubelet

mikedanese committed:
Create a service account Getter when TokenRequest is enabled

Fixes a nil panic when --service-account-lookup=false and TokenRequest
is enabled.

k8s-ci-robot committed:
Merge pull request #79480 from wojtek-t/fix_clientgo_benchmark

Fix client-go request benchmark

k8s-ci-robot committed:
Merge pull request #73977 from khenidak/ipv6dualstack

ipv6 dual stack (Phase 1 - ALPHA)

k8s-ci-robot committed:
Merge pull request #72974 from sjenning/sysctl-kubeconfig

kubelet: add allowed sysctl to KubeletConfiguration

k8s-ci-robot committed:
Merge pull request #79675 from mikedanese/safix

Create a service account Getter when TokenRequest is enabled

k8s-ci-robot committed:
Merge pull request #79391 from danwinship/nodeaddresses-update-fix

Don't use strategic merge patch on Node.Status.Addresses

k8s-ci-robot committed:
Merge pull request #73982 from tanshanshan/fixcp

fix cp time stamp is in the future error

BenTheElder committed:
correct flag order in make-help

mborsz committed:
Add MASTER_NODE_LABELS

k8s-ci-robot committed:
Merge pull request #79701 from BenTheElder/gnu-warns

correct flag order in make-help

mborsz committed:
Fix HA setup logic

k8s-ci-robot committed:
Merge pull request #79713 from mborsz/ha

Fix HA setup logic

k8s-ci-robot committed:
Merge pull request #78789 from johscheuer/remove-experimental-note-for-ipvs

Remove experimental note for ipvs in kube-proxy arguments

k8s-ci-robot committed:
Merge pull request #79703 from mborsz/master_node_labels

Add MASTER_NODE_LABELS

yastij committed:
consolidate coreDNS version to ease dependency bump

Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

k8s-ci-robot committed:
Merge pull request #79270 from SataQiu/fix-kubeadm-2019062102

kubeadm: add --control-plane-endpoint flag

k8s-ci-robot committed:
Merge pull request #79705 from qingsenLi/k8s-190703

correct the CHANGELOG-1.12.md

k8s-ci-robot committed:
Merge pull request #72420 from Pingan2017/cleanup-outofdisk

clean up redundant conditiontype ‘OutOfDisk’

k8s-ci-robot committed:
Merge pull request #79045 from odinuge/file-perms

Fix file permissions for non executable files

k8s-ci-robot committed:
Merge pull request #79529 from andyzhangx/change-csi-timeout

fix: change default timeout value in csi plugin

Commit:
correct the CHANGELOG-1.12.md

k8s-ci-robot committed:
Merge pull request #77819 from smarterclayton/client

Add a metadata client to client-go that can read PartialObjectMetadata

k8s-ci-robot committed:
Merge pull request #79728 from yastij/consolidate-coredns-versions

consolidate coreDNS version to ease dependency bump

tiffanyfay committed:
Remove LazyProvide from cloud providers

tiffanyfay committed:
Remove LazyProvide from credential provider keyring and provider
